lib/src/screens/send/send_page.dart

```diff
@@ -48,10 +49,42 @@ class SendFormState extends State<SendForm> {
   final _cryptoAmountController = TextEditingController();
   final _fiatAmountController = TextEditingController();
 
+  final _focusNode = FocusNode();
+
   bool _effectsInstalled = false;
 
   final _formKey = GlobalKey<FormState>();
 
+  @override
+  void initState() {
+    _focusNode.addListener(() async {
+      if (!_focusNode.hasFocus &&_addressController.text.isNotEmpty) {
+        await fetchXmrAddressAndRecipientName(formatDomainName(_addressController.text)).then((map) async {
```

Comment:
Why not move this logic into send store ? And there check is entered text is domain or xmr address and then if it is domain fetch needed info there ?